web/satellite/src/views/register/Register.vue

```diff
@@ -46,7 +46,15 @@
                 this.secret = this.$route.query.token.toString();
             }
 
-            let { ids } = this.$route.params;
+            let { ids = '' } = this.$route.params;
+            let decoded = '';
+            try {
+                decoded = atob(ids);
+            } catch {
+                this.$router.push(ROUTES.LOGIN);
+
+                return;
+            }
             let referralIds = ids ? JSON.parse(atob(ids)) : undefined;
```

Comment:
let referralIds = ids ? JSON.parse(decoded) : undefined;

not sure if it make sense to call atob(ids) twice

Reply:
fixed it